src/video_core/shader/generator/glsl_shader_gen.cpp

```diff
@@ -300,15 +329,39 @@ void PicaGSConfigCommonRaw::Init(const Pica::Regs& regs) {
             if (static_cast<std::size_t>(semantic) < 24) {
                 semantic_maps[static_cast<std::size_t>(semantic)] = {attrib, comp};
             } else if (semantic != VSOutputAttributes::INVALID) {
-                LOG_ERROR(Render_OpenGL, "Invalid/unknown semantic id: {}", semantic);
+                LOG_ERROR(Render, "Invalid/unknown semantic id: {}", semantic);
             }
         }
     }
 }
 
-PicaFixedGSConfig::PicaFixedGSConfig(const Pica::Regs& regs, const Instance& instance) {
-    state.Init(regs);
-    use_clip_planes = instance.IsShaderClipDistanceSupported();
+void PicaVSConfigState::Init(const Pica::Regs& regs, Pica::Shader::ShaderSetup& setup,
+                             bool use_clip_planes_, bool use_geometry_shader_) {
+    use_geometry_shader = use_geometry_shader_;
+
+    program_hash = setup.GetProgramCodeHash();
+    swizzle_hash = setup.GetSwizzleDataHash();
+    main_offset = regs.vs.main_offset;
+    sanitize_mul = VideoCore::g_hw_shader_accurate_mul;
+
+    num_outputs = 0;
+    load_flags.fill(AttribLoadFlags::Float);
+    output_map.fill(16);
+
+    for (int reg : Common::BitSet<u32>(regs.vs.output_mask)) {
+        output_map[reg] = num_outputs++;
+    }
+
+    gs_state.Init(regs, use_clip_planes_);
```

Comment:
```suggestion
    if (!use_geometry_shader) {
        gs_state.Init(regs, use_clip_planes_);
    }
```
This is only used when geometry shaders are unavailable